src/java/main/org/apache/zookeeper/ZooKeeper.java

```diff
@@ -1436,7 +1436,7 @@ public String create(final String path, byte data[], List<ACL> acl,
         request.setData(data);
         request.setFlags(createMode.toFlag());
         request.setPath(serverPath);
-        if (acl != null && acl.size() == 0) {
+        if (acl == null || acl.isEmpty() || acl.contains(null)) {
```

Comment:
These validations should also be applied on async create APIs 

1. org.apache.zookeeper.ZooKeeper.create(String, byte[], List<ACL>, CreateMode, Create2Callback, Object, long) 
2. org.apache.zookeeper.ZooKeeper.create(String, byte[], List<ACL>, CreateMode, StringCallback, Object)